src/Provider/DatabaseServiceProvider.php

```diff
@@ -81,8 +84,16 @@ function ($app) {
             $app->extend(
                 'db',
                 function ($db) use ($app) {
+                    /** @var \Bolt\Storage\Database\Connection|\Bolt\Storage\Database\MasterSlaveConnection $db */
                     $db->setQueryCacheProfile($app['db.query_cache_profile']);
 
+                    // Currently, this can get called early, meaning the connection listener is too late
+                    if (!Type::hasType('json')) {
+                        Type::addType('json', Schema\Types\JsonType::class);
+                    }
```

Comment:
Maybe we should override it for 2.6+ so it is consistent? Our Type is different than the one doctrine provides.

Reply:
In that case we might as well lock on DBAL 2.5, and I'd just have to maintain a fork Bolt for PHP 7.1+. But I don't think the differences for 2.5 users are enough of a difference to warrant it.

Reply:
How does that follow? I am saying ours now matches what dbal 3.0 will look like. So we need to apply it to all of dbal 2.x

Reply:
First thing that comes to mind is the exception and JSON processing … there were others, but at this point if that's a breaker I am just deleting the branch